pkg/pillar/types/zedkubetypes.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import "time"
+
+// KubeNodeStatus - Enum for the status of a Kubernetes node
+type KubeNodeStatus int8
+
+const (
+	KubeNodeStatusUnknown      KubeNodeStatus = iota // KubeNodeStatusUnknown - Node status is unknown
+	KubeNodeStatusReady                              // KubeNodeStatusReady - Node is in ready status
+	KubeNodeStatusNotReady                           // KubeNodeStatusNotReady - Node is in not ready status
+	KubeNodeStatusNotReachable                       // KubeNodeStatusNotReachable - Node is not reachable
+)
+
+// KubeNodeInfo - Information about a Kubernetes node
+type KubeNodeInfo struct {
+	Name               string
+	Status             KubeNodeStatus
+	IsMaster           bool
+	IsEtcd             bool
+	CreationTime       time.Time
+	LastTransitionTime time.Time
+	KubeletVersion     string
+	InternalIP         string
```

Comment:
I would suggest using net.IP type for IP addresses.

Reply:
We got this from kubernetes 'nodestatus' struct, which is a list of string for 'Adddress'.

Reply:
OK